pyanaconda/modules/network/initialization.py

```diff
@@ -240,89 +240,97 @@ def _run(self, nm_client):
                 # Filter out potenital connection created for BOOTIF option rhbz#2175664
-                port_cons = [c for c in cons if not is_bootif_connection(c)]
-                if len(port_cons) == 1 and self._is_initramfs_connection(port_cons[0], iface):
+                port_cons = [c for c in available_cons if not is_bootif_connection(c)]
+                if len(port_cons) == 1 and initramfs_cons:
                     log.debug("%s: device %s has an initramfs port connection",
                               self.name, iface)
-                    con = self._select_persistent_connection_for_device(
+                    dumped_con = self._select_persistent_connection_for_device(
                         device, port_cons, allow_ports=True)
                 else:
                     log.debug("%s: creating default connection for port device %s",
                               self.name, iface)
 
-            if not con:
-                con = self._select_persistent_connection_for_device(device, cons)
-
-            has_initramfs_con = any(self._is_initramfs_connection(con, iface) for con in cons)
-            if has_initramfs_con:
-                log.debug("%s: device %s has initramfs connection", self.name, iface)
-                if not con and n_cons == 1:
-                    # Try to clone the persistent connection for the device
-                    # from the connection which should be a generic (not bound
-                    # to iface) connection created by NM in initramfs
-                    con = clone_connection_sync(nm_client, cons[0], con_id=iface)
-
-            if con:
-                self._update_connection(nm_client, con, iface)
-                # Update some values of connection generated in initramfs so it
-                # can be used as persistent configuration.
-                if has_initramfs_con:
-                    update_connection_values(
-                        con,
-                        [
-                            # Make sure ONBOOT is yes
-                            (NM.SETTING_CONNECTION_SETTING_NAME,
-                             NM.SETTING_CONNECTION_AUTOCONNECT,
-                             True),
-                            # Update cloned generic connection from initramfs
-                            (NM.SETTING_CONNECTION_SETTING_NAME,
-                             NM.SETTING_CONNECTION_MULTI_CONNECT,
-                             0),
-                            # Update cloned generic connection from initramfs
-                            (NM.SETTING_CONNECTION_SETTING_NAME,
-                             NM.SETTING_CONNECTION_WAIT_DEVICE_TIMEOUT,
-                             -1)
-                        ]
-                    )
+            if not dumped_con:
+                dumped_con = self._select_persistent_connection_for_device(device, available_cons)
+
+            if not dumped_con and len(initramfs_cons) == 1:
```

Comment:
I think you can put the second if under the first one and remove the `not dumped_con` part of the condition.